### PR TITLE
Fixing release build in Dev6x

### PR DESCRIPTION
**Pull request by jmprieur**
# Fixes the release build in dev6x

### CODE REVIEW

File: test/Microsoft.IdentityModel.AotCompatibility.Tests/Microsoft.IdentityModel.AotCompatibility.Tests.csproj [hunk 1]
```diff
@@ -5,6 +5,7 @@
   <PropertyGroup>
     <!-- This test only needs to run on .NET -->
     <TargetFrameworks>net6.0</TargetFrameworks>
+    <LangVersion>10</LangVersion>
```
Comment (FuPingFranco) [inline test/Microsoft.IdentityModel.AotCompatibility.Tests/Microsoft.IdentityModel.AotCompatibility.Tests.csproj:8]: nit: should we keep this consistent with commonTest.props by using 11?

File: build/common.props [hunk 2]
```diff
@@ -34,9 +34,19 @@
...
+    <EnablePackageValidation>true</EnablePackageValidation>
+    <PackageValidationBaselineVersion>6.33.0</PackageValidationBaselineVersion>
+  </PropertyGroup>
+
   <PropertyGroup Condition="'$(TF_BUILD)' == 'true'">
     <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
   </PropertyGroup>
+
+  <PropertyGroup>
+    <NoWarn>$(NoWarn);SYSLIB0050</NoWarn>
```
Comment (FuPingFranco) [inline build/common.props:47]: Do we have a follow up task to deal with these warnings If we suppress now? or is that not a concern?

File: updateAssemblyInfo.ps1 [hunk 3]
```diff
@@ -43,7 +43,7 @@ Write-Host "assemblyInformationalVersion: "  $assemblyInformationalVersion
 $nugetSuffix = [string]$buildConfiguration.SelectSingleNode("root/nugetSuffix").InnerText
 if ( $packageType -eq "release")
 {
-    $versionSuffix = ""
+    $versionSuffix = $nugetSuffix
```
Comment (sruke) [inline updateAssemblyInfo.ps1]: nugetSuffix is typically set to preview in [buildConfiguration.xml]. By making this change, versionSuffix would be set to preview for release packages. Should Line 46 be reverted?

**Review by sruke — COMMENTED** [review]
Reviewed with one open question. Everything else LGTM!

File: build/common.props (hunk 2 above)
Comment (westin-m) [reply build/common.props:47]: It's something that has to be kept to maintain support for older target frameworks. Such a task could come up when we stop supporting whichever tf needs it.

**Review by westin-m — APPROVED** [review]

File: build/common.props (hunk 2 above)
Comment (FuPingFranco) [reply build/common.props:47]: Sounds good, resolving this comment.

File: updateAssemblyInfo.ps1 (hunk 3 above)
Comment (jmprieur) [reply updateAssemblyInfo.ps1]: do you mind taking care of this, @sruke?

File: test/Microsoft.IdentityModel.AotCompatibility.Tests/Microsoft.IdentityModel.AotCompatibility.Tests.csproj (hunk 1 above)
Comment (jmprieur) [reply test/Microsoft.IdentityModel.AotCompatibility.Tests/Microsoft.IdentityModel.AotCompatibility.Tests.csproj:8]: We would, but this does not work with net6.0

Comment (jmprieur) [reply test/Microsoft.IdentityModel.AotCompatibility.Tests/Microsoft.IdentityModel.AotCompatibility.Tests.csproj:8]: We could also remove the targetframeworks and this line

Comment (FuPingFranco) [reply test/Microsoft.IdentityModel.AotCompatibility.Tests/Microsoft.IdentityModel.AotCompatibility.Tests.csproj:8]: I see, resolving this comment. Thanks Jean-Marc!

File: updateAssemblyInfo.ps1 (hunk 3 above)
Comment (sruke) [reply updateAssemblyInfo.ps1]: Set versionSuffix back to an empty string for release packages

**Review by brentschmaltz — APPROVED** [review]

**Review by brentschmaltz — CHANGES_REQUESTED** [review]
@jmprieur  We need to skip TestExtensions for API compat

**Comment by jmprieur** [comment]
I know, @brentschmaltz. I've spend my week-end disabling it in the fork ...

**Pull request closed without merging**